### ISSUE TITLE
latest: pod without the network annotation stuck in ContainerCreating

### ISSUE THREAD

**Issue by mumoshu** [k8snetworkplumbingwg/multus-cni#642]
TL;DR: https://github.com/k8snetworkplumbingwg/multus-cni/commit/8d7308e6bb613e83e22767c8affc9cc8a08d36eb#r48925504

cc/ @s1061123

**What happend**:

Any pod without the network annotation seems to stuck in ContainerCreating:

```
NAME   READY   STATUS              RESTARTS   AGE
pod1   0/1     ContainerCreating   0          31m
```

**What you expected to happen**:

Pods without the network annotation gets created and be running as before(v3.7.1 or lesser)

**How to reproduce it (as minimally and precisely as possible)**:

- `kind create cluster` (I've used kind 0.8.1
- Deploy `kube-multus-ds` with the latest multus-cni container image with the config below:
  ```
  {
      "cniVersion": "0.3.1",
      "name": "multus-cni-network",
      "type": "multus",
      "logLevel": "debug",
      "logFile": "/var/log/multus.log",
      "confDir": "/etc/cni/net.d",
      "clusterNetwork": "kindnet",
      "defaultNetworks": [],
      "systemNamespaces": [""],
      "kubeconfig": "/etc/cni/net.d/multus.d/multus.kubeconfig"
    }
  ```
- Create a network attachment definition for `kindnet` as usual
   ```
  apiVersion: "k8s.cni.cncf.io/v1"
  kind: NetworkAttachmentDefinition
  metadata:
    name: kindnet
    namespace: kube-system
  ```
- Create a pod without the network annotation:
  ```
  kind: Pod
  apiVersion: v1
  metadata:
    name: pod1
  spec:
    containers:
    - name: pause
      image: k8s.gcr.io/pause:3.1
  ```

**Anything else we need to know?**:

I believe this was caused by this commit: https://github.com/k8snetworkplumbingwg/multus-cni/commit/8d7308e6bb613e83e22767c8affc9cc8a08d36eb#r48925504

Also note that the same scenario just work with v3.7.1.

**Environment**:

- Multus version `latest` (Please see below)
  image path and image ID (from 'docker images')

> ```
> root@kind-control-plane:/# ctr -n k8s.io images ls | grep latest
> ghcr.io/k8snetworkplumbingwg/multus-cni:lates
> application/vnd.docker.distribution.manifest.list.v2+json sha256:63b2a1cb91892b1b4308d6615c46490c57d4b1c103b62d69b97609e66bbb302b 100.0 MiB linux/amd64,linux/arm64,linux/ppc64le,linux/s390x           io.cri-containerd.image=managed
> ```

- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.0", GitCommit:"af46c47ce925f4c4ad5cc8d1fca46c7b77d13b38", GitTreeState:"clean", BuildDate:"2020-12-08T17:59:43Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-21T01:11:42Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
```
- Primary CNI for Kubernetes cluster:

`kind`'s default network `ptp + portmap`

- OS (e.g. from /etc/os-release):
- File of '/etc/cni/net.d/'
- File of '/etc/cni/multus/net.d'
- NetworkAttachment info (use `kubectl get net-attach-def -o yaml`)
- Target pod yaml info (with annotation, use `kubectl get pod <podname> -o yaml`)
- Other log outputs (if you use multus logging)
